[17.0][MIG] web_widget_product_label_section_and_note: Migration to version 17.0

carlos-lopez-tecnativa opened the pull request:
- Removed the `resizeTextArea` function, as it is now included in Odoo 17.
- `resizeInput` is not exportable, so it still needs to be kept. See https://github.com/odoo/odoo/blob/30f7928ac42c8536a6179b96ed7f8aa9d0f14374/addons/web/static/src/core/utils/autoresize.js#L42
- `useAutoresize` is available in Odoo 17, but it does not support passing an `onResize` function in the `options`. Therefore, the existing code must be retained as well.
V17
https://github.com/odoo/odoo/blob/30f7928ac42c8536a6179b96ed7f8aa9d0f14374/addons/web/static/src/core/utils/autoresize.js#L26
V18
https://github.com/odoo/odoo/blob/3757daca84dc080a276c04f310262a3ddf340a0f/addons/web/static/src/core/utils/autoresize.js#L27

TT54841
@Tecnativa @pedrobaeza @chienandalu @legalsylvain could you please review this.

carlos-lopez-tecnativa committed:
[ADD] web_widget_product_label_section_and_note: New module to unify the product and name into a single column.

carlos-lopez-tecnativa committed:
[IMP] web_widget_product_label_section_and_note: Auto-resize textarea for multiline text.
Before this change, when the text was large and multiline, the textarea did not resize, and the text was not displayed correctly. Now, the textarea adjusts according to the text. These changes are backported from V18.

mymage committed:
Added translation using Weblate (Italian)

carlos-lopez-tecnativa committed:
[IMP] web_widget_product_label_section_and_note: pre-commit auto fixes

carlos-lopez-tecnativa committed:
[MIG] web_widget_product_label_section_and_note: Migration to version 17.0

pedrobaeza commented:
/ocabot migration web_widget_product_label_section_and_note

pedrobaeza commented:
Tested functionally in the base DB, as the other is very contaminated by things like the classic theme.

This problem was present also on 16.0 version, but let's tackle it (and maybe backport the fix): the "not show" the product name is not preserved when changing the rest of the text:

![Peek 05-02-2025 20-19](https://github.com/user-attachments/assets/986b2e73-3cf3-4760-871c-5d0d2c63f8f1)

About the migration itself, 2 comments:

- The grey background color for opening the product is normal?
- On the sales order, the widget is not working the same as it did in 16. Please check.

carlos-lopez-tecnativa commented:
> About the migration itself, 2 comments:
> 
> * The grey background color for opening the product is normal?

I updated the style to V17, following the many2one style.

> * On the sales order, the widget is not working the same as it did in 16. Please check.

I remember that in sales, it did not work if the `sale_product_configurator` module was installed(the `ROADMAP` states this). I deactivated the view on Runboat, and now it is working. Please check again. Additionally, I remember that you changed this module to `auto_install=False`, but it is still installed on Runboat. I don’t know why.
https://github.com/OCA/OCB/pull/1242/files#diff-6d2599689bebadcbba9c676b8656305600bcdcf0c2b359f57cdef023136ef065R29

carlos-lopez-tecnativa committed:
[FIX] web_widget_product_label_section_and_note: Use a flag to change the icon based on product visibility.
Before this commit, when isProductVisible was disabled and the user changed the label, this caused isProductVisible to change its value.
After this commit, changing the label no longer modifies isProductVisible.

carlos-lopez-tecnativa commented:
> This problem was present also on 16.0 version, but let's tackle it (and maybe backport the fix): the "not show" the product name is not preserved when changing the rest of the text:
> 

I added a separate commit to fix this. Please try again and let me know if it is okay for you, and if it can be backported to V16.

pedrobaeza reviewed (approved):
Tested and working. You can backport the latest fix.

pedrobaeza commented:
/ocabot merge nobump
